graph/src/ipfs/gateway_client.rs

```diff
@@ -353,108 +257,13 @@ mod tests {
             .unwrap();
     }
 
-    #[tokio::test]
-    async fn new_does_not_retry_gateway_check_on_non_retriable_errors() {
-        let server = mock_server().await;
-
-        mock_gateway_check(StatusCode::METHOD_NOT_ALLOWED)
-            .expect(1)
-            .mount(&server)
-            .await;
-
-        IpfsGatewayClient::new(server.uri(), &discard())
-            .await
-            .unwrap_err();
-    }
-
     #[tokio::test]
     async fn new_unchecked_creates_the_client_without_checking_the_gateway() {
         let server = mock_server().await;
 
         IpfsGatewayClient::new_unchecked(server.uri(), &discard()).unwrap();
     }
 
-    #[tokio::test]
```

Comment:
did you replace these with something else?

Reply:
These tests were for the `.can_provide()` method used by the IPFS client pool to select the appropriate client for requests, but this is no longer needed, so the method and its tests have been removed.

I also removed tests for non-retriable errors, because from the perspective of IPFS clients, all errors are now retriable, except for deterministic errors that are not implemented yet.